Gems/EMotionFX/Code/Source/Editor/SkeletonModel.cpp

```diff
@@ -234,9 +268,52 @@ namespace EMotionFX
         Node* node = static_cast<Node*>(index.internalPointer());
         AZ_Assert(node, "Expected valid node pointer.");
 
+        const bool isRootNode = node == m_characterRootNode;
         const size_t nodeIndex = node->GetNodeIndex();
-        const NodeInfo& nodeInfo = m_nodeInfos[nodeIndex];
+        const NodeInfo& nodeInfo = GetNodeInfo(node);
 
+        // Handle roles for the special case with a root node
+        if (isRootNode)
+        {
+            switch (role)
+            {
+            case Qt::DecorationRole:
+                {
+                    switch (index.column())
+                    {
+                    case COLUMN_RAGDOLL_LIMIT:
+                        [[fallthrough]];
+                    case COLUMN_RAGDOLL_COLLIDERS:
+                        [[fallthrough]];
+                    case COLUMN_HITDETECTION_COLLIDERS:
+                        [[fallthrough]];
+                    case COLUMN_CLOTH_COLLIDERS:
+                        [[fallthrough]];
+                    case COLUMN_SIMULATED_JOINTS:
+                        [[fallthrough]];
+                    case COLUMN_SIMULATED_COLLIDERS:
```

Comment:
suggestion: I think all the `fallthroughs` here aren't required and add a lot of noise, when you have no logic at all in each case just omitting is fine (also it might be nice to extract this logic to a function too so it's not just appearing in the larger function)